### PR TITLE
Cat D flow for waiting room complete

### PR DESCRIPTION
**Pull request by a0dexx**
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/waiting-room/cat-d/__tests__/waiting-room.cat-d.page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+import { LockScreenIndicator } from '../../../../components/common/screen-lock-indicator/lock-screen-indicator';
+import { CandidateSectionComponent } from '../../../../components/common/candidate-section/candidate-section';
+import { FormControl, Validators } from '@angular/forms';
+import { candidateMock } from '../../../../modules/tests/__mocks__/tests.mock';
+import { CatDUniqueTypes } from '@dvsa/mes-test-schema/categories/D';
+import { App } from '../../../../app/app.component';
+import { MockAppComponent } from '../../../../app/__mocks__/app.component.mock';
+import { configureTestSuite } from 'ng-bullet';
+
+describe('WaitingRoomCatCPage', () => {
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-d/__tests__/waiting-room.cat-d.page.spec.ts]: Can we fix the name please

File: src/pages/page-names.constants.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ export const CAT_C = {
   PASS_FINALISATION_PAGE: 'PassFinalisationCatCPage',
   REVERSE_DIAGRAM_PAGE: 'ReverseDiagramCatCPage',
   TEST_REPORT_PAGE: 'TestReportCatCPage',
-  WAITING_ROOM_PAGE: 'WaitingRoomCatCPage',
+  WAITING_ROOM_PAGE: 'WaitingRoomCatDPage',
```
Comment (TomBillingtonUK) [inline src/pages/page-names.constants.ts]: Undo this please as this is a Cat C Page

File: src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+  candidateName$: Observable<string>;
+  candidateUntitledName$: Observable<string>;
+  candidateDriverNumber$: Observable<string>;
+  welshTest$: Observable<boolean>;
+  conductedLanguage$: Observable<string>;
+}
+
+@IonicPage()
+@Component({
+  selector: '.waiting-room-cat-c-page',
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts]: needs changing to d

File: src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+  }
+
+  dispatchCandidateChoseToProceedInEnglish() {
+    this.store$.dispatch(new CandidateChoseToProceedWithTestInEnglish(Language.ENGLISH));
+  }
+
+  onSubmit() {
+    Object.keys(this.formGroup.controls).forEach(controlName => this.formGroup.controls[controlName].markAsDirty());
+    if (this.formGroup.valid) {
+      this.navController.push(CAT_C.COMMUNICATION_PAGE);
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts]: Should be CAT_D

**Review by TomBillingtonUK — APPROVED** [review]
Happy to approve as long as above comments are addressed

File: src/pages/page-names.constants.ts (hunk 2 above)
Comment (a0dexx) [reply src/pages/page-names.constants.ts]: done

File: src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts (hunk 3 above)
Comment (a0dexx) [reply src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts]: done

File: src/pages/page-names.constants.ts (hunk 2 above)
Comment (a0dexx) [reply src/pages/page-names.constants.ts]: done

File: src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts (hunk 4 above)
Comment (a0dexx) [reply src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts]: done

File: src/pages/waiting-room/cat-d/__tests__/waiting-room.cat-d.page.spec.ts (hunk 1 above)
Comment (a0dexx) [reply src/pages/waiting-room/cat-d/__tests__/waiting-room.cat-d.page.spec.ts]: done

File: src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+  getPreTestDeclarations,
+} from '../../../modules/tests/pre-test-declarations/common/pre-test-declarations.reducer';
+import * as preTestDeclarationsActions
+  from '../../../modules/tests/pre-test-declarations/common/pre-test-declarations.actions';
+import {
+  getInsuranceDeclarationStatus,
+  getResidencyDeclarationStatus,
+  getSignatureStatus,
+} from '../../../modules/tests/pre-test-declarations/common/pre-test-declarations.selector';
+import { getCandidate } from '../../../modules/tests/journal-data/cat-c/candidate/candidate.cat-c.reducer';
```
Comment (mgoldson) [inline src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts:19]: How are tackling instances where we have to reference other categories for now? Sticking with the TODO approach or something different?

**Review by mgoldson — APPROVED** [review]
Good work Anthony! A question to consider and can you add a description for the PR please? Especially useful for cross team context without going into the JIRA ticket.

File: src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts (hunk 5 above)
Comment (a0dexx) [reply src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts:19]: i guess TODO approach.  But working through the sub tasks for ticket 4291 we these issues will be resolved one by one

**Pull request merged**